### ISSUE TITLE
Markdown containing horizontal rows breaks parser

### ISSUE THREAD

**Issue by leplastic** [Modularr/YAML-FrontMatter#21]
If you load a document, for example:

    ---
    title: Some title
    date: 2017-05-15
    ---
    This is a markdown document separated by horizontal rows:
    ---
    There is one above!

The FrontMatter will try to split the document by ``---``, and then gives Symphony's YAML the 2nd position of array, which is in this case "This is a markdown (...)", instead of the desired section. If you check the [markdown reference](http://daringfireball.net/projects/markdown/syntax) or the [Github flavored mardown reference](https://help.github.com/articles/organizing-information-with-tables/), you will see that ``---`` appears in other options, such as headers and tables. So when a document includes such formatting, it will cause trouble on FrontMatter.

I hope I'm not being too annoying by opening a new issue :) I use FrontMatter on some projects and it is also of my interest to see your project doing well.

**Comment by bmcminn** [comment]
Have you tried adding more hyphens to your break line? I usually do 5+ since my syntax highlighter colors it differently that way, but it's still valid markdown.

**Comment by Blaxus** [comment]
I changed the way it was interpreted when I ported to a new Yaml parser (in reverse), but I didn't really think about dashes inside of markdown, Which is what's likely what is causing this. I'll take a look soon.

**Comment by leplastic** [comment]
@bmcminn I tried now, but it's the same result. My editor (Atom) starts highlighting at 3 characters.

I don't use this markup often, but since I have a test Markdown document to test the Mardown engine I'm using, I noticed problems using that particular file with FrontMatter. I will then wait for the fix. This version is very good because the inclusion of Symphony's YAML, now I can use Dates and Numbers instead of doing manual conversions from Strings.

**Comment by Blaxus** [comment]
OKay I've basically for now reset back to the old code. I really didn't think too much about the splitting of the blocks and figuring out a "cleaner" way is something I don't have time for right now. I should have really thought about that before I went trying to refactor.

I've tested your code. It should work, I've basically copied the old code, and replaced all the yaml parsers including my own shitty one with Symfony Yaml. So as far as compatibility goes it should be all fixed now. Update to 1.3.2
